### PR TITLE
Feat/add histogram for continuous covariates

### PR DESCRIPTION
**Pull request by pieterlukasse**
Jira Ticket: [VADC-79](https://ctds-planx.atlassian.net/browse/VADC-79)

### New Features
- Added new histogram component that connects with cohort-middleware histogram endpoint
- Small fix to Euler diagram storybook

Screenshot:

Screenshot 2022-12-16 at 12 24 55

### CODE REVIEW

File: src/Analysis/GWASV2/Components/Diagrams/PhenotypeHistogram/Histogram.jsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import {
+  BarChart, Bar, XAxis, YAxis, CartesianGrid, Tooltip
+} from 'recharts';
+import PropTypes from 'prop-types';
+
+//TODO - add legend, auto-scaling, sorting, etc
```
Comment (pieterlukasse) [inline src/Analysis/GWASV2/Components/Diagrams/PhenotypeHistogram/Histogram.jsx:7]: future ticket.

Comment (jarvisraymond-uchicago) [reply src/Analysis/GWASV2/Components/Diagrams/PhenotypeHistogram/Histogram.jsx]: Would you like me to create a ticket for this? Or have you created one?

Comment (pieterlukasse) [reply src/Analysis/GWASV2/Components/Diagrams/PhenotypeHistogram/Histogram.jsx]: I'll add one

File: src/Analysis/GWASV2/Components/Diagrams/PhenotypeHistogram/Histogram.stories.jsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  title: "Tests3/GWASV2/Histogram_Recharts",
+  component: Histogram,
+};
+
+const Template = (args) => <Histogram {...args} />;
+
+export const FirstStepActive = Template.bind({});
+FirstStepActive.args = {
+  data: [
+    { ageBin: '1-2', patients: 10000 },
```
[the remark this answers is not in the document]
Comment (pieterlukasse) [reply src/Analysis/GWASV2/Components/Diagrams/PhenotypeHistogram/Histogram.stories.jsx:14]: yes, this is just testing/showing how recharts will handle such scenarios where x-axis item is a mix of types. It basically just treats everything as strings, which is ok.

File: src/Analysis/GWASV2/Components/Diagrams/PhenotypeHistogram/PhenotypeHistogram.stories.jsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        //histogram/by-source-id/${sourceId}/by-cohort-definition-id/${cohortId}/by-histogram-concept-id/${currentSelection.concept_id}`;
+        'http://:cohortmiddlewarepath/cohort-middleware/histogram/by-source-id/:sourceid/by-cohort-definition-id/:cohortdefinitionId/by-histogram-concept-id/:conceptId',
+        (req, res, ctx) => {
+          const { cohortmiddlewarepath } = req.params;
+          const { cohortdefinitionId } = req.params;
+          return res(
+            ctx.delay(1100),
+            ctx.json({
+              bins: [
+                {start: 1,  end: 10, nr_persons: 100},
```
[the remark this answers is not in the document]
Comment (pieterlukasse) [reply src/Analysis/GWASV2/Components/Diagrams/PhenotypeHistogram/PhenotypeHistogram.stories.jsx:79]: yes, `nr_persons` is the number of persons (`person` records - see CDM data model - e.g. https://github.com/uc-cdis/cohort-middleware/commit/bb6c8f0467803ecd089d7fc1575a29f8c55a68d8#diff-772602eb720bef605d9da3beeff5bc06335b32e4a530ff5d1f5cc6b5a6f6c2b8R22)) in the bin

File: src/Analysis/GWASV2/Components/Diagrams/PhenotypeHistogram/PhenotypeHistogram.jsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  if (status === 'error') {
+    return <React.Fragment>Error getting data for diagram</React.Fragment>;
+  }
+  if (status === 'loading') {
+    return <Spin />;
+  }
+  const histogramArgs = {
+    data: data.bins,
+    xAxisDataKey: 'start',
+    barDataKey: 'nr_persons',
```
[the remark this answers is not in the document]
Comment (pieterlukasse) [reply src/Analysis/GWASV2/Components/Diagrams/PhenotypeHistogram/PhenotypeHistogram.jsx:45]: it is a count (number of persons in the bin). In the context of a histogram it is OK naming IMO, but we can change to `number_of_person_records` to make it more explicit. Let me know if you prefer that. That would require a change in the backend as well.

File: src/Analysis/GWASV2/Components/Diagrams/PhenotypeHistogram/PhenotypeHistogram.stories.jsx [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+                {start: 10, end: 20, nr_persons: 200},
```
[the remark this answers is not in the document]
Comment (pieterlukasse) [reply src/Analysis/GWASV2/Components/Diagrams/PhenotypeHistogram/PhenotypeHistogram.stories.jsx:80]: I'm not sure yet if we want the ordering to happen in the frontend. Maybe it is easier to just do it in the backend, and let the frontend just render in the order it receives it... I think cohort-middleware already returns it ordered, so adding a sorting here in the FE is a nice to have to make the component more generic for other use cases...we can tackle that when such a use case comes around.

File: src/Analysis/GWASV2/Components/Diagrams/PhenotypeHistogram/PhenotypeHistogram.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (jarvisraymond-uchicago) [reply src/Analysis/GWASV2/Components/Diagrams/PhenotypeHistogram/PhenotypeHistogram.jsx:45]: Ok I prefer pronounceable variable names (Robert Martin's idea, https://bpoplauschi.github.io/2021/01/30/Clean-Code-Naming-by-Uncle-Bob-part-3.html#make-sure-names-are-pronounceable). I added this to the renaming things ticket here: https://ctds-planx.atlassian.net/browse/VADC-348

**Pull request merged**